src/v/cluster/topic_table.cc

```diff
@@ -1103,7 +1142,9 @@ topic_properties topic_table::update_topic_properties(
       updated_properties.iceberg_delete, overrides.iceberg_delete);
     incremental_update(
       updated_properties.iceberg_partition_spec,
-      overrides.iceberg_partition_spec);
+      overrides.iceberg_partition_spec,
+      std::optional(
+        config::shard_local_cfg().iceberg_default_partition_spec()));
```

Comment:
q: what's the purpose of explicitly invoking `std::optional`'s constructor here?

Reply:
Otherwise the overload resolution won't work - topic property is an optional, but the config property is not (and the function expects them both to be the same type), so it has to be converted to an optional as well.

Reply:
> the function expects them both to be the same type

ah, didn't notice that. thx